helm/templates/storage/deployment.yaml

```diff
@@ -23,3 +23,10 @@ spec:
         - name: storage
           image: {{ .Values.storage.image.repository }}:{{ .Values.storage.image.tag }}
           imagePullPolicy: {{ .Values.storage.image.pullPolicy }}
+          volumeMounts:
+            - name: sqlite-pvc
+              mountPath: /data/sqlite/
+      volumes:
+        - name: sqlite-pvc
+          persistentVolumeClaim:
+            claimName: {{ .Values.storage.pvc.name }}
```

Comment:
To be handled differently, see for example what is being done [here](https://github.com/nextcloud/helm/blob/00d8a86ba8bab3f10b7180946dfe37615a071f4c/charts/nextcloud/templates/deployment.yaml#L350-L354).

Something along these lines:
```yaml
  - name: data-sqlite
    persistentVolumeClaim:
            claimName: {{ if .Values.persistence.storageData.existingClaim }}{{ .Values.persistence.storageData.existingClaim }}{{- else }}{{ template "storage.fullname" . }}-sqlite-data{{- end }}
```

Double check what the `template staroge.fullname` does.